docs/apache-airflow/concepts/dynamic-task-mapping.rst

```diff
@@ -0,0 +1,256 @@
+
+This can be thought of as defining your tasks in a for loop, but instead of having to have the DAG file fetch the data and do that itself, the scheduler can do this based on the output of a previous task. Right before a mapped task is executed the scheduler will create *n* copies of the task, one for each input.
+
+It is also possible to have a task operate on the collected output of a mapped task, commonly known as map and reduce.
+
+Simple mapping
+==============
+
+In its simplest form you can map over a list defined directly in your DAG file using the ``expand()`` function instead of calling your task directly.
+
+.. code-block:: python
+
+    from airflow import DAG
+    from airflow.decorators import task
+
+
+    with DAG(dag_id="simple_mapping", start_date=datetime(2022, 3, 4)):
+
+        @task
+        def add_one(x: int):
+            return x + 1
+
+        @task
+        def sum_it(values):
+            total = sum(values)
+            print(f"Total was {total}")
+
+        added_values = add_one.expand(x=[1, 2, 3])
+        sum_it(added_values)
+
+This will show ``Total was 9`` in the task logs when executed.
+
+.. note:: A reduce task is not required.
+
+    Although we show a "reduce" task here (``sum_it``) you don't have to have one, the mapped tasks will still be executed even if they have no downstream tasks.
+
+Repeated Mapping
+================
+
+The result of one mapped task can also be used as input to the next mapped task
+
+.. code-block:: python
+
+    from airflow import DAG
+    from airflow.decorators import task
+
+
+    with DAG(dag_id="repeated_mapping", start_date=datetime(2022, 3, 4)):
+
+        @task
+        def add_one(x: int):
+            return x + 1
+
+        first = add_one.expand(x=[1, 2, 3])
+        second = add_one.expand(x=first)
+
+This would have a result of [3, 4, 5]
+
+Constant parameters
+===================
+
+As well as passing arguments that get expanded at run-time, it is possible to pass arguments that don't change – in order to clearly differentiate between the two kinds we use different functions, ``expand()`` for mapped arguments, and ``partial()`` for unmapped ones.
+
+For example:
+
+.. code-block:: python
+
+    @task
+    def add(x: int, y: int):
+        return x + y
+
+
+    added_values = add.partial(y=10).expand(x=[1, 2, 3])
+    # This results in add function being expanded to
+    # add(x=1,y=10)
+    # add(x=2,y=10)
+    # add(x=3,y=10)
+
+This would result in values of 11, 12, 13.
+
+This is also useful for passing things such as connection IDs, database table names, or bucket names to tasks.
+
+Mapping over multiple parameters
+================================
+
+As well as a single parameter it is possible to pass multiple parameters to expand. This will have the effect of creating a "cross product" effect, calling the mapped task with each combination of parameters.
```

Comment:
```suggestion
As well as a single parameter it is possible to pass multiple parameters to expand. This will have the effect of creating a "cross product", calling the mapped task with each combination of parameters.
```